spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1112,12 +1112,7 @@ private RuntimeException doInvokeRecordListener(final ConsumerRecord<K, V> recor
 						this.errorHandler.handle(e, record, this.consumer);
 					}
 					if (producer != null) {
-						try {
-							sendOffsetsToTransaction(producer);
-						}
-						catch (Exception e1) {
-							this.logger.error("Send offsets to transaction failed", e1);
-						}
+						ackCurrent(record, producer);
```

Comment:
I wonder why the block in the beginning of the `catch` doesn't work:
```
if (this.containerProperties.isAckOnError() && !this.autoCommit && producer == null) {
		ackCurrent(record, producer);
}
```

That's one of the option the reporter in the #683 uses.

May you think what's wrong is there?

Thanks

Reply:
Because the producer is not null.

Reply:
We can't do this up there because we need to wait for the error handler to process before we decide to commit or roll back.